packages/expect-puppeteer/src/matchers/toMatch.js

```diff
@@ -1,12 +1,23 @@
-async function toMatch(page, matcher, options = { timeout: 500 }) {
+async function toMatch(handler, matcher, options = { timeout: 500 }) {
+  const handlerType = handler.constructor.name;
+
+  if (handlerType === 'ElementHandle') {
+    const textContentProperty = await handler.getProperty('textContent');
+    const textJson = await textContentProperty.jsonValue();
+
+    if (!textJson.match(matcher)) throw  Error(`Expected to find text: "${matcher}"`);
```

Comment:
Can we use the same strategy? Use a `waitForFunction`?

Your code expect text synchronously, the current `expect(page).toMatch` waits 500ms.

Reply:
We could potentially move this into a `waitForFunction`. The idea behind this pull request was to utilize the text access process that @aslushnikov outlined [here](https://github.com/GoogleChrome/puppeteer/issues/438#issuecomment-340675046). 

> Your code expect text synchronously
 
This method will just try and match on whatever `ElementHandle` is passed to the `expect`. It does assume a user has passed a *"loaded"* element. From a programmatical standpoint it isn't a synchronous "blocking" call, it is just won't wait (in its current implementation) for the allotted time using the `options.timeout`. 

If that is indeed the issue, I'd be happy to brainstorm with you about the right way to solve this.

Reply:
OK for the element, for me it is OK we assume that it is loaded. It is not the business of the function.

My point is that for `expect(element).toMatch`, it text is not in the element it will not wait. Whereas for `expect(page).toMatch` it will wait.

This two piece of code should have the same behavior:

```js
// First
triggerAsyncRequestThatPopulateBody()
const body = await page.$('body')
await expect(body).toMatch('Some text') // this will fail

// Second
triggerAsyncRequestThatPopulateBody()
await expect(page).toMatch('Some text') // this will work
```

For me "First" and "Second" should both work. We can easily add a test for it.